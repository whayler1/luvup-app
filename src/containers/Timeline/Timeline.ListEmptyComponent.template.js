import React from 'react';
import { View, Text } from 'react-native';

import { scene, vars } from '../../styles';
import Well from '../../components/Well';
import Preloader from '../../components/Preloader';

const ListEmptyComponent = ({ isInFlight, error }) => (
  <View
    style={{
      paddingHorizontal: 16,
      paddingVertical: 32,
      alignSelf: 'stretch',
    }}>
    {error && (
      <Well text="There was an error loading the timeline. Make sure you are connected to wifi or data." />
    )}
    {isInFlight && <Preloader />}
    {!error &&
      !isInFlight && (
        <Text style={[scene.copy, { fontSize: 25, textAlign: 'center' }]}>
          You do not have any history yet. Send a Luvup to get things started!
        </Text>
      )}
  </View>
);

export default ListEmptyComponent;
