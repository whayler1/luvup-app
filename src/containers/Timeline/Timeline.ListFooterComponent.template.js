import React from 'react';
import { View } from 'react-native';

import styles from './Timeline.styles';
import Preloader from '../../components/Preloader';

export default ({ isPresent, isPreloaderVisible }) => {
  if (!isPresent) {
    return false;
  }
  return (
    <View style={styles.listFooterContainer}>
      {isPreloaderVisible && <Preloader />}
    </View>
  );
};
