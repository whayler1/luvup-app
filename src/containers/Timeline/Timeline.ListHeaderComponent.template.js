import React from 'react';
import { View, Text } from 'react-native';
import { Group, Surface } from 'ReactNativeART';
import Color from 'color';

import Circle from '../../components/Circle';
import { vars } from '../../styles';
import styles from './Timeline.styles';
import CoinArt from '../../components/CoinArt';
import JalapenoArt from '../../components/JalapenoArt';

const coinStrokeColor = Color(vars.razzleDazzleRose).mix(Color('white'), 0.3);

const leaderboardSlot = ({
  relationshipScore,
  initials,
  coinCount,
  jalapenoCount,
  isSecond,
}) => (
  <View
    style={isSecond ? styles.leaderboardSlotSecond : styles.leaderboardSlot}>
    <View style={styles.leaderboardSlotUserWrapper}>
      <Surface width={44} height={44}>
        <Group x={1} y={1}>
          <Circle radius={21} stroke="rgba(255,255,255,0.7)" strokeColor={2} />
        </Group>
      </Surface>
      <Text style={styles.leaderboardSlotInitials}>{initials}</Text>
    </View>
    <View style={styles.leaderboardSlotStatsWrapper}>
      <View style={styles.leaderboardSlotScoreWrapper}>
        <Text style={styles.leaderboardSlotScore}>
          {`${relationshipScore}%`}
        </Text>
      </View>
      <CoinArt
        fill="rgba(255,255,255,0.6)"
        stroke={coinStrokeColor}
        scale={0.3}
      />
      <Text style={styles.leaderboardText}>{coinCount}</Text>
      <View style={styles.leaderboardSlotJalapenoArtWrapper}>
        <JalapenoArt fill="rgba(255,255,255,0.6)" scale={0.25} />
      </View>
      <Text style={styles.leaderboardText}>{jalapenoCount}</Text>
    </View>
  </View>
);

export default ({
  userRelationshipScore,
  loverRelationshipScore,
  userInitials,
  loverInitials,
  coinCount,
  jalapenoCount,
  sentCoinsCount,
  sentJalapenosCount,
}) => (
  <View style={styles.listHeaderContainer}>
    {leaderboardSlot({
      relationshipScore: userRelationshipScore,
      initials: userInitials,
      coinCount,
      jalapenoCount,
    })}
    {leaderboardSlot({
      relationshipScore: loverRelationshipScore,
      initials: loverInitials,
      coinCount: sentCoinsCount,
      jalapenoCount: sentJalapenosCount,
      isSecond: true,
    })}
  </View>
);
