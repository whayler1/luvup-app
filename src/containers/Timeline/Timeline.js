import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import superagent from 'superagent';
import moment from 'moment';
import _ from 'lodash';

import {
  setUserEvents as setUserEventsAction,
  getUserEvents as getUserEventsAction,
  clearUserEvents as clearUserEventsAction,
} from '../../redux/userEvents/userEvents.actions';
import { setSentCoinsCount as setSentCoinsCountAction } from '../../redux/coin/coin.actions';
import { setSentJalapenosCount as setSentJalapenosCountAction } from '../../redux/jalapeno/jalapeno.actions';
import config from '../../config';
import Template from './Timeline.template';

const userEventsLimit = 100;
const format = 'YYYY-MM-DD';

const getSections = userEvents => {
  let currentCreatedDate;
  let currentEventName;
  return userEvents.reduce((val, event, i) => {
    const eventCreatedDate = moment(new Date(event.createdAt)).format(format);
    if (eventCreatedDate !== currentCreatedDate) {
      currentCreatedDate = eventCreatedDate;
      currentEventName = undefined;
      val.push({
        title: moment(currentCreatedDate).format('ddd, MMM DD, YYYY'),
        data: [],
      });
    }
    if (event.name !== currentEventName) {
      currentEventName = event.name;
      val[val.length - 1].data.push({
        ...event,
        key: event.id,
        time: moment(new Date(event.createdAt)).format('h:mma'),
        count: 1,
      });

    } else {
      const lastVal = val[val.length - 1];
      const lastEvt = lastVal.data[lastVal.data.length - 1];
      lastEvt.count++;
    }

    return val;
  }, []);
};

class Timeline extends Component {
  static propTypes = {
    userFirstName: PropTypes.string,
    userLastName: PropTypes.string,
    loverFirstName: PropTypes.string,
    loverLastName: PropTypes.string,
    coinCount: PropTypes.number,
    jalapenoCount: PropTypes.number,
    sentCoinsCount: PropTypes.number,
    sentJalapenosCount: PropTypes.number,
    userEvents: PropTypes.array,
    userEventsCount: PropTypes.number,
    setUserEvents: PropTypes.func.isRequired,
    getUserEvents: PropTypes.func.isRequired,
    clearUserEvents: PropTypes.func.isRequired,
    setSentCoinsCount: PropTypes.func.isRequired,
    setSentJalapenosCount: PropTypes.func.isRequired,
  };

  state = {
    prevMostRecentUserEvent: undefined,
    sections: [],
    isSectionsLoaded: false,
    page: 0,
    isModalVisible: false,
    isAtEndOfList: false,
  }

  closeModal = () => this.setState({ isModalVisible: false });

  getMoreUserEvents = async () => {
    const res = await this.props.getUserEvents(userEventsLimit, this.state.page * userEventsLimit, true);
  }

  onEndReached = _.throttle(() => {
    /**
     * - do not do anything if sections havent loaded yet.
     * - do not do anything if # items loaded is equal to or more then count
     */
    if (this.state.isSectionsLoaded &&
        (userEventsLimit * (this.state.page + 1)) < this.props.userEventsCount) {
      this.setState({
        page: this.state.page + 1,
      }, this.getMoreUserEvents);
    } else {
      this.setState({ isAtEndOfList: true });
    }
  }, 250);

  goToDashboard = () => Actions.dashboard();

  setSections = () => {
    const sections = getSections(this.props.userEvents);
    /**
     * JW: we have to stagger "isSectionsLoaded" to happen one frame after
     * adding sections so the sections can draw before we fire onEndReached
     */
    this.setState({ sections },
      () => setTimeout(() => this.setState({ isSectionsLoaded: true }), 500));
  };

  setInitials = () => {
    const {
      userFirstName,
      userLastName,
      loverFirstName,
      loverLastName,
    } = this.props;

    this.setState({
      userInitials: (userFirstName.substr(0,1) + userLastName.substr(0,1)).toUpperCase(),
      loverInitials: (loverFirstName.substr(0,1) + loverLastName.substr(0,1)).toUpperCase(),
    });
  };

  componentWillMount = async () => {
    this.setInitials();

    const query = `{
      userEvents(
        limit: ${userEventsLimit}
        offset: 0
      ) {
        rows {
          id isViewed createdAt name
        }
        count
      }
      sentCoins(limit: 0) { count }
      sentJalapenos(limit: 0) { count }
    }`;

    try {
      const res = await superagent.post(config.graphQlUrl, { query });

      const { userEvents, sentCoins, sentJalapenos } = res.body.data;
      const { setUserEvents, setSentCoinsCount, setSentJalapenosCount } = this.props;

      setUserEvents(userEvents.rows, userEvents.count);
      setSentCoinsCount(sentCoins.count);
      setSentJalapenosCount(sentJalapenos.count);

      this.setSections();
    } catch (err) {
      this.setState({ isModalVisible: true });
      console.log('Timeline componentWillMount err', err);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.userEvents.length && nextProps.userEvents[0].createdAt !== this.state.prevMostRecentUserEvent) {
      this.setSections();
    }
  }

  render() {
    return <Template
      {...this.props}
      {...this.state}
      goToDashboard={this.goToDashboard}
      onEndReached={this.onEndReached}
      closeModal={this.closeModal}
    />;
  }
}

export default connect(
  state => ({
    userFirstName: state.user.firstName,
    userLastName: state.user.lastName,
    loverFirstName: state.lover.firstName,
    loverLastName: state.lover.lastName,
    coinCount: state.coin.count,
    jalapenoCount: state.jalapeno.count,
    sentCoinsCount: state.coin.sentCoinsCount,
    sentJalapenosCount: state.jalapeno.sentJalapenosCount,
    userEvents: state.userEvents.rows,
    userEventsCount: state.userEvents.count,
  }),
  {
    setUserEvents: setUserEventsAction,
    getUserEvents: getUserEventsAction,
    clearUserEvents: clearUserEventsAction,
    setSentCoinsCount: setSentCoinsCountAction,
    setSentJalapenosCount: setSentJalapenosCountAction,
  }
)(Timeline);
