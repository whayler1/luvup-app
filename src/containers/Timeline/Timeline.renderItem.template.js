import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { Actions } from 'react-native-router-flux';
import Icon from 'react-native-vector-icons/Ionicons';
import _ from 'lodash';

import styles from './Timeline.styles';
import { vars } from '../../styles';

import NotificationDot from '../../components/NotificationDot';
import jalapenoSentImg from '../../images/jalapeno-sent.png';
import jalapenoReceivedImg from '../../images/jalapeno-received.png';
import coinSentImg from '../../images/coin-sent.png';
import coinReceivedImg from '../../images/coin-received.png';
import LoveNoteArtFlying from '../../components/LoveNoteArtFlying';
import LoveNoteArtReceived from '../../components/LoveNoteArtReceived';

const getEventDisplayName = (eventName, count) => {
  const plur = count > 1 ? 's' : '';

  switch (eventName) {
    case 'coin-sent':
      return `Luvup${plur} sent`;
    case 'coin-received':
      return `Luvup${plur} received`;
    case 'jalapeno-sent':
      return `Jalapeno${plur} sent`;
    case 'jalapeno-received':
      return `Jalapeno${plur} received`;
    case 'password-changed':
      return 'Password Changed';
    case 'lovenote-sent':
      return `Love note${plur} sent`;
    case 'lovenote-received':
      return `Love note${plur} received`;
    default:
      return eventName;
  }
};
const getEventImage = eventName => {
  switch (eventName) {
    case 'coin-sent':
      return (
        <Image
          style={{
            width: 32,
            height: 25,
          }}
          source={coinSentImg}
        />
      );
    case 'coin-received':
      return (
        <Image
          style={{
            width: 31,
            height: 25,
          }}
          source={coinReceivedImg}
        />
      );
    case 'jalapeno-sent':
      return (
        <Image
          style={{
            width: 24,
            height: 25,
          }}
          source={jalapenoSentImg}
        />
      );
    case 'jalapeno-received':
      return (
        <Image
          style={{
            width: 26,
            height: 25,
          }}
          source={jalapenoReceivedImg}
        />
      );
    case 'password-changed':
      return <Icon name="md-lock" size={36} color={vars.blueGrey500} />;
    case 'lovenote-sent':
      return (
        <View style={{ paddingTop: 3 }}>
          <LoveNoteArtFlying scale={0.4} />
        </View>
      );
    case 'lovenote-received':
      return (
        <View style={{ paddingTop: 3 }}>
          <LoveNoteArtReceived scale={0.4} />
        </View>
      );
    default:
      return;
  }
};

const Wrapper = ({ children, eventName, loveNote = {} }) => {
  if (_.isString(loveNote.id)) {
    return (
      <TouchableOpacity
        onPress={() => Actions.viewLoveNote({ loveNoteId: loveNote.id })}>
        {eventName === 'lovenote-received' &&
          !loveNote.isRead && (
            <NotificationDot
              style={{
                position: 'absolute',
                left: 12,
                top: 10,
              }}
            />
          )}
        {...children}
      </TouchableOpacity>
    );
  }
  return <View>{...children}</View>;
};

export default ({ item, index, section }) => {
  const isLovenoteItem = /^lovenote/.test(item.name);
  const isLovenoteItemWithNote =
    isLovenoteItem && _.isString(_.get(item, 'loveNote.id'));

  return (
    <Wrapper
      isLink={isLovenoteItemWithNote}
      loveNote={item.loveNote}
      eventName={item.name}>
      <View
        style={
          index + 1 === section.data.length
            ? styles.renderItemContainerLast
            : styles.renderItemContainer
        }>
        <View style={styles.renderItemContent}>
          <View
            style={{
              minWidth: 50,
            }}>
            {getEventImage(item.name)}
          </View>
          <View
            style={{
              paddingLeft: 16,
              paddingRight: 16,
              flex: 1,
            }}>
            <Text style={styles.renderItemContentText}>
              {item.name !== 'password-changed' && !isLovenoteItem
                ? item.count + ' '
                : ''}
              {getEventDisplayName(item.name, item.count)}
            </Text>
            {isLovenoteItem &&
              _.isString(_.get(item, 'loveNote.note')) && (
                <Text
                  numberOfLines={1}
                  style={{
                    color: vars.blueGrey500,
                    fontSize: 14,
                    fontFamily: vars.fontRegular,
                    paddingTop: 8,
                  }}>
                  {decodeURI(item.loveNote.note)}
                </Text>
              )}
          </View>
          <View>
            <Text style={styles.renderItemContentSmall}>{item.time}</Text>
          </View>
        </View>
      </View>
    </Wrapper>
  );
};
