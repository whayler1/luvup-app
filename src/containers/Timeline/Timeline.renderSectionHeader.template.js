import React from 'react';
import { View, Text } from 'react-native';

import { vars } from '../../styles';
import styles from './Timeline.styles';

export default ({ section }) => (
  <View style={styles.sectionHeaderContainer}>
    <Text style={styles.sectionHeaderText}>{section.title}</Text>
  </View>
);
