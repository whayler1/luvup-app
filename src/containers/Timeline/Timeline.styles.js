import { StyleSheet } from 'react-native';
import { vars } from '../../styles';

const renderItemContainer = {
  paddingTop: 16,
  paddingLeft: 16,
  paddingRight: 16,
  paddingBottom: 8,
  alignSelf: 'stretch',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
};
const renderItemContainerLast = {
  ...renderItemContainer,
  paddingBottom: 24,
};
const leaderboardSlot = {
  paddingLeft: 16,
  paddingRight: 16,
  flexDirection: 'row',
  alignItems: 'center',
};
const leaderboardSlotSecond = {
  ...leaderboardSlot,
  paddingTop: 16,
};

export default new StyleSheet.create({
  wrapper: {
    flex: 1,
    alignItems: 'center',
  },
  sectionList: {
    alignSelf: 'stretch',
  },
  listHeaderContainer: {
    paddingBottom: 24,
    backgroundColor: vars.razzleDazzleRose,
  },
  listFooterContainer: {
    paddingBottom: 16,
    paddingTop: 16,
  },
  leaderboardSlot,
  leaderboardSlotSecond,
  leaderboardText: {
    marginLeft: 8,
    fontFamily: vars.fontBlack,
    fontSize: 20,
    color: vars.blueGrey500,
  },
  sectionHeaderContainer: {
    paddingTop: 0,
    paddingLeft: 16,
    paddingRight: 16,
    paddingBottom: 16,
    borderBottomColor: vars.blueGrey100,
    borderBottomWidth: 1,
    alignSelf: 'stretch',
    backgroundColor: 'white',
  },
  sectionHeaderText: {
    color: vars.blueGrey800,
    fontSize: 20,
    fontFamily: vars.fontBlack,
  },
  renderItemContainer,
  renderItemContainerLast,
  renderItemContent: {
    flexDirection: 'row',
    flex: 1,
  },
  renderItemContentText: {
    color: vars.blueGrey700,
    fontSize: 20,
    fontFamily: vars.fontRegular,
    flex: 1,
  },
  renderItemContentSmall: {
    color: vars.blueGrey500,
    fontSize: 14,
    fontFamily: vars.fontRegular,
    paddingTop: 4,
  },
  renderItemLoveNoteText: {
    color: vars.blueGrey300,
    fontSize: 14,
    fontFamily: vars.fontBlack,
    paddingTop: 4,
  },
  renderItemLoveNoteTokenRow: {
    flexDirection: 'row',
    marginTop: 8,
  },
  renderItemLoveNoteTokenItem: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  renderItemLoveNoteTokenText: {
    fontFamily: vars.fontBlack,
    fontSize: 14,
    color: vars.blueGrey500,
    paddingLeft: 2,
  },
  heartBtn: {
    flex: 1,
    alignItems: 'flex-end',
  },
  heartImg: {
    width: 32,
    height: 30,
  },
  sectionListWrapper: {
    paddingTop: 83,
    alignSelf: 'stretch',
  },
});
