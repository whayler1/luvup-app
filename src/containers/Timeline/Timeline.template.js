import React from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  SectionList,
  Modal,
  RefreshControl,
} from 'react-native';
import { Button } from 'react-native-elements';

import styles from './Timeline.styles';
import { buttons, scene, modal } from '../../styles';
import renderItem from './Timeline.renderItem.template';
import renderSectionHeader from './Timeline.renderSectionHeader.template';
import ListHeaderComponent from './Timeline.ListHeaderComponent.template';
import ListFooterComponent from './Timeline.ListFooterComponent.template';
import ListEmptyComponent from './Timeline.ListEmptyComponent.template';
import HeartArt from '../../components/Art/HeartArt';

export default ({
  userRelationshipScore,
  loverRelationshipScore,
  coinCount,
  jalapenoCount,
  sentCoinsCount,
  sentJalapenosCount,
  goBack,
  sections,
  isAfterFirstLoad,
  isSectionsLoaded,
  userInitials,
  loverInitials,
  onRefresh,
  onEndReached,
  isModalVisible,
  closeModal,
  isGetUserEventsInFlight,
  isGetTimelineDataInFlight,
  getUserEventsError,
}) => (
  <View style={styles.wrapper}>
    <View style={[scene.topNav, styles.topNav]}>
      <TouchableOpacity onPress={goBack} style={styles.heartBtn}>
        <HeartArt scale={0.037} fill="rgba(0,0,0,0.5)" />
      </TouchableOpacity>
    </View>
    <View style={styles.sectionListWrapper}>
      <SectionList
        endFillColor="white"
        style={styles.sectionList}
        refreshControl={
          <RefreshControl
            style={styles.refreshControl}
            enabled={
              isAfterFirstLoad &&
              !isGetUserEventsInFlight &&
              !isGetTimelineDataInFlight
            }
            refreshing={
              isAfterFirstLoad &&
              (isGetUserEventsInFlight || isGetTimelineDataInFlight)
            }
            onRefresh={onRefresh}
            tintColor="white"
          />
        }
        ListEmptyComponent={
          <ListEmptyComponent
            isInFlight={isGetUserEventsInFlight}
            error={getUserEventsError}
          />
        }
        ListHeaderComponent={
          <ListHeaderComponent
            {...{
              userRelationshipScore,
              loverRelationshipScore,
              coinCount,
              jalapenoCount,
              userInitials,
              loverInitials,
              sentCoinsCount,
              sentJalapenosCount,
            }}
          />
        }
        ListFooterComponent={
          <ListFooterComponent
            isPresent={sections.length > 0}
            isPreloaderVisible={isGetUserEventsInFlight && isSectionsLoaded}
          />
        }
        renderSectionHeader={renderSectionHeader}
        renderItem={renderItem}
        sections={sections}
        onEndReached={onEndReached}
      />
    </View>
    <Modal visible={isModalVisible} animationType={'fade'} transparent>
      <View style={modal.outerContainer}>
        <View style={modal.innerContainer}>
          <Text style={modal.title}>Error</Text>
          <Text style={modal.copy}>
            There was an error loading your timeline. Most likely this is due to
            network conectivity.
          </Text>
          <View style={modal.buttonContainer}>
            <Button
              raised
              onPress={closeModal}
              containerViewStyle={buttons.infoContainer}
              buttonStyle={buttons.infoButton}
              textStyle={buttons.infoText}
              title="Dismiss"
            />
          </View>
        </View>
      </View>
    </Modal>
  </View>
);
