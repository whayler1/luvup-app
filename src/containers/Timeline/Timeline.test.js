import React from 'react';
import renderer from 'react-test-renderer';

import Timeline from './Timeline';
import store from '../../redux';

it('renders without crashing', () => {
  const rendered = renderer.create(<Timeline store={store} />).toJSON();
  expect(rendered).toBeTruthy();
  expect(rendered).toMatchSnapshot();
});
