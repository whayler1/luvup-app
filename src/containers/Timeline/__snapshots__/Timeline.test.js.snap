// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "left": 0,
          "marginTop": 28,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 16,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 10,
        },
        Object {
          "backgroundColor": "#f94dda",
          "marginTop": 0,
          "paddingTop": 45,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <RNSVGSvgView
        bbHeight="29.858999999999998"
        bbWidth="32.967"
        height={29.858999999999998}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 29.858999999999998,
              "width": 32.967,
            },
          ]
        }
        width={32.967}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              0,
              0,
              0,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={Array []}
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth="1"
          x={0}
          y={0}
        >
          <RNSVGPath
            d="M445.386719,806.233245 L445,806.617188 L445,805.849302 L68.109375,431.664062 C22.703125,386.033854 -1.0236633e-13,321.8125 -1.42108547e-14,239 C1.27495206e-13,114.78125 124.789062,2.50949378e-14 253,-3.55271368e-15 C338.218345,-1.94921149e-14 402.34725,25.9638009 445.386715,77.8914026 C488.426184,25.9638023 552.55509,-1.94921153e-14 637.773438,-3.55271368e-15 C765.984375,2.50949378e-14 890.773438,114.78125 890.773438,239 C890.773438,321.8125 868.070313,386.033854 822.664063,431.664062 L445.773438,805.849302 L445.773438,806.617188 L445.386719,806.233245 Z"
            fill={
              Array [
                0,
                0,
                0,
                0,
                0.5,
              ]
            }
            fillOpacity={1}
            fillRule={1}
            matrix={
              Array [
                0.037,
                0,
                0,
                0.037,
                0,
                0,
              ]
            }
            opacity={1}
            originX={0}
            originY={0}
            propList={
              Array [
                "fill",
              ]
            }
            rotation={0}
            scaleX={0.037}
            scaleY={0.037}
            skewX={0}
            skewY={0}
            stroke={null}
            strokeDasharray={null}
            strokeDashoffset={null}
            strokeLinecap={0}
            strokeLinejoin={0}
            strokeMiterlimit={4}
            strokeOpacity={1}
            strokeWidth="1"
            x={0}
            y={0}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "paddingTop": 83,
      }
    }
  />
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.7)",
          "flex": 1,
          "justifyContent": "center",
          "padding": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#CFD8DC",
            "borderRadius": 2,
            "borderWidth": 1,
            "paddingBottom": 32,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 32,
            "shadowColor": "#607D8B",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#607D8B",
              "fontFamily": "quicksandbold",
              "fontSize": 30,
              "textAlign": "center",
            }
          }
        >
          Error
        </Text>
        <Text
          style={
            Object {
              "color": "#607D8B",
              "fontFamily": "quicksandregular",
              "fontSize": 20,
              "marginTop": 16,
            }
          }
        >
          There was an error loading your timeline. Most likely this is due to network conectivity.
        </Text>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "marginTop": 32,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                Object {
                  "shadowColor": "rgba(0,0,0, .4)",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 1,
                },
                Object {
                  "alignSelf": "stretch",
                  "marginLeft": 0,
                  "marginRight": 0,
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "#4DD0E1",
                      "borderRadius": 2,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        Object {
                          "fontFamily": "quicksandbold",
                          "textAlign": "center",
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Dismiss
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
